core/whistle_number_manager-1.0.0/src/wh_number_manager.erl

```diff
@@ -1007,7 +1007,10 @@ prepare_find_results([Number|Numbers], ModuleName, ModuleResults, Found, Opts) -
                       | Found]
              end,
     case catch wnm_number:get(Number) of
-        #number{state=State} ->
+        #number{state=State
+                ,module_data=JObj
+                ,module_name = ModuleName
```

Comment:
@lazedo 
Correct me if I am wrong, but it looks like the discovery is only re-saved if the module_name does not match?  IE: if a search is preformed by the same carrier module then it can not resave the number if it already exists because it will match this case-clause (which is the old behavior).

If so lets talk about it :)